### PR TITLE
Add option to ignore pods with PVCs from eviction

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/kubernetes-sigs/descheduler/issues/96

### CODE REVIEW

**Comment by damemi** [comment]
/hold
`make gen` is currently broken due to the logging fields

**Review by damemi — COMMENTED** [review]
/hold cancel
make gen is fixed now

File: cmd/descheduler/app/options/options.go [hunk 1]
```diff
@@ -77,4 +77,5 @@ func (rs *DeschedulerServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&rs.MaxNoOfPodsToEvictPerNode, "max-pods-to-evict-per-node", rs.MaxNoOfPodsToEvictPerNode, "DEPRECATED: limits the maximum number of pods to be evicted per node by descheduler")
 	// evict-local-storage-pods allows eviction of pods that are using local storage. This is false by default.
 	fs.BoolVar(&rs.EvictLocalStoragePods, "evict-local-storage-pods", rs.EvictLocalStoragePods, "DEPRECATED: enables evicting pods using local storage by descheduler")
+
```
Comment (ingvagabund) [inline cmd/descheduler/app/options/options.go:80]: You can drop the additional line

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Can you add a test case for the new flag? In any of the strategies you choose.

Wrt. `ignore` prefix, we already have `ExcludeOwnerKinds` and `Exclude` namespaces fields. Do you think it would make sense to align the new field with existing ones? I.e. `excludePVCPods`? Even `IgnorePVCPods` sounds ok to me.

**Review by lixiang233 — COMMENTED** [review]
Besides, we should update the [user guide](https://github.com/kubernetes-sigs/descheduler/blob/master/docs/user-guide.md) to include the new flag

**Comment by damemi** [comment]
Updated this to remove the extra line and added the flag to readme

**Comment by ingvagabund** [comment]
Unit test please

**Comment by damemi** [comment]
Sorry, forgot the unit test. Added to `PodLifeTime`

**Review by lixiang233 — APPROVED** [review]
/lgtm

**Comment by jsravn** [comment]
Minor nit: could we rename the option from `ignorePvcPods` to `evictPvcPods` before this is released? I found the option name confusing - I took it to mean "ignore PVCs when considering a pod for eviction", rather than "ignore pods with PVCs when evicting".

**Comment by damemi** [comment]
Hi @jsravn, thanks for the feedback. The motivation behind naming this `ignorePvcPods` was the the current default Descheduler behavior is to evict Pvc Pods. So, rather than add the new flag with a default value `true`, we added it so that it semantically defaults to `false`, preserving the existing default behavior without any change to behavior.

I hope this clears up the confusion. At this point, I would personally prefer not to make that change since it would be a full invert of the behavior. However if others disagree I am open to the consideration

**Pull request merged**